File: src/Building Blocks/Ark.Shared/Entities/Entity.cs
using System;

namespace Ark.Core.Entities
{
    public abstract class Entity
    {
        public Guid Id { get; private set;  }

        public DateTime CreatedAt { get; private set; }
        public DateTime UpdatedAt { get; private set; }
        public DateTime DeletedAt { get; private set; }


        public Entity()
        {
            this.Id = Guid.NewGuid();
            this.CreatedAt = DateTime.UtcNow;
        }
    }
}

File: src/domain/entities/core/entity.py
from datetime import datetime
from typing import Any, Dict, Type
from uuid import UUID, uuid4

import pydantic
from beanie import Document, SaveChanges, before_event


class Entity(Document):
    created_at: datetime
    updated_at: datetime

    @before_event(SaveChanges)
    def update_timestamps(self):
        self.updated_at = datetime.utcnow()

    class Config:
        arbitrary_types_allowed = True
